app/design-system/design-system-internals/src/main/kotlin/com/hedvig/android/design/system/internals/SnackBar.kt

```diff
@@ -0,0 +1,158 @@
+package com.hedvig.android.design.system.internals
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.material3.Snackbar
+import androidx.compose.material3.SnackbarHost
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.SnackbarResult.ActionPerformed
+import androidx.compose.material3.SnackbarResult.Dismissed
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun InternalSnackBar(
+  snackbarText: String,
+  showSnackbar: Boolean,
+  showedSnackbar: () -> Unit,
+  colors: NotificationColors,
+  shape: Shape,
+  textStyle: TextStyle,
+  icon: @Composable () -> Unit,
+  modifier: Modifier = Modifier,
+  action: (() -> Unit)? = null,
+  actionLabel: String? = null,
+) {
+  val snackbarHostState = remember { SnackbarHostState() }
+  LaunchedEffect(showSnackbar, snackbarText) {
+    if (!showSnackbar) return@LaunchedEffect
+    val result = snackbarHostState.showSnackbar(snackbarText, actionLabel = actionLabel)
+    when (result) {
+      Dismissed -> {
+        showedSnackbar()
+      }
+
+      ActionPerformed -> {
+        if (action != null) {
+          action()
+        }
+        showedSnackbar()
+      }
+    }
+  }
+  SnackbarHost(snackbarHostState, modifier.padding(16.dp)) { _ ->
+    Snackbar(
+      action = {
+        if (actionLabel != null) {
```

Comment:
I don't know if this is relevant here or not, but the M3 API for this looks like it accepts a `action: @Composable (() -> Unit)?,` which means it would take `null` too,
Now, in the case where we have no actionLabel, we are passing it an empty lambda instead of a null lambda.
This *may* have implications on how it measures some stuff internally.
Do we perhaps want to try passing null here instead when we do not want to have an action and make sure things look OK like that too?